crypto/bn/bn_test.cc

```diff
@@ -655,50 +654,6 @@ static bool TestHex2BN() {
   return true;
 }
 
-static bool TestRand(RAND *rng) {
-  ScopedBIGNUM bn(BN_new());
-  if (!bn) {
-    return false;
-  }
-
-  // Test BN_rand accounts for degenerate cases with |top| and |bottom|
-  // parameters.
-  if (!BN_rand(bn.get(), 0, 0 /* top */, 0 /* bottom */, rng) ||
-      !BN_is_zero(bn.get())) {
-    fprintf(stderr, "BN_rand gave a bad result.\n");
-    return false;
-  }
```

Comment:
I agree, it is unfortunate. But, the test is already there so it's just a matter of deleting less of it. The test for `BN_rand` partially establishes the validity of the `BN_bn2bin_padded` test. Ideally we'd rewrite the `BN_bn2bin_padded` test "soonish" but I kind of suspect it's likely to stay like it is for a while as it's "just a test" and so likely not very urgent to everybody. Especially since we've already remove `BN_rand_range` and `BN_rand` from the main lib, which is the main win.